### PR TITLE
add vectorAdd example

### PR DESCRIPTION
**Pull request by vincentridder**
i added a simple cuplafied example from cuda sdk: vectorAdd

### CODE REVIEW

**Comment by bussmann** [comment]
Nice

File: example/CUDASamples/vectorAdd/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#
+# Copyright 2016 Rene Widera, Benjamin Worpitz
```
Comment (psychocoderHPC) [inline example/CUDASamples/vectorAdd/CMakeLists.txt]: please add you name add the end

**Comment by psychocoderHPC** [comment]
Could you please squash some of the commits to one. So that we had two or three commits in the end.

File: example/CUDASamples/vectorAdd/src/vectorAdd.cpp [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+ * CUDA Kernel Device code
+ *
+ * Computes the vector addition of A and B into C. The 3 vectors have the same
+ * number of elements numElements.
+ */
+struct vectorAdd {
+    template<typename T_Acc>
+    ALPAKA_FN_HOST_ACC
+    void operator()(T_Acc const &acc, const float *A, const float *B, float *C, const int numElements) const {
+        int begin = blockDim.x * blockIdx.x * elemDim.x + threadIdx.x;
```
Comment (psychocoderHPC) [inline example/CUDASamples/vectorAdd/src/vectorAdd.cpp]: This is only working if `blockDim = dim3(1,1,1)`.
Should be:
```C++
 int begin = blockDim.x * blockIdx.x * elemDim.x + threadIdx.x * elemDim.x;
```

File: example/CUDASamples/vectorAdd/src/vectorAdd.cpp [hunk 3]
```diff
@@ -35,7 +35,7 @@ struct vectorAdd {
     template<typename T_Acc>
     ALPAKA_FN_HOST_ACC
     void operator()(T_Acc const &acc, const float *A, const float *B, float *C, const int numElements) const {
-        int begin = blockDim.x * blockIdx.x * elemDim.x + threadIdx.x;
+        int begin = blockDim.x * blockIdx.x * elemDim.x + threadIdx.x*elemDim.x;
```
Comment (psychocoderHPC) [inline example/CUDASamples/vectorAdd/src/vectorAdd.cpp]: space missing before `* elemDim.x`

**Comment by psychocoderHPC** [comment]
Thanks for the new example, good work!

**Pull request merged**